libs/vkd3d-shader/spirv.c

```diff
@@ -11141,9 +11141,36 @@ static bool is_dcl_instruction(enum VKD3D_SHADER_INSTRUCTION_HANDLER handler_idx
             || handler_idx == VKD3DSIH_HS_DECLS;
 }
 
+static bool is_scope_generating_control_flow_instruction(enum VKD3D_SHADER_INSTRUCTION_HANDLER handler)
+{
+    /* Ignore continue, break, ret and any unconditional branches which don't logically generate new labels on their own.
+     * In some Xenia shaders, there is questionable back-to-back continue + continue, followed by break.
+     * If we've terminated a block, just ignore it.
+     * Technically, if there are if/endif pairs after a continue or break, it might be more
+     * correct to track shadow control flow scopes and ignore code until we pop out of the shadow stack,
+     * but this is silly and should only be considered if actually needed. */
+    switch (handler)
+    {
+        case VKD3DSIH_CASE:
+        case VKD3DSIH_DEFAULT:
+        case VKD3DSIH_ELSE:
+        case VKD3DSIH_ENDIF:
+        case VKD3DSIH_ENDLOOP:
+        case VKD3DSIH_ENDSWITCH:
+        case VKD3DSIH_IF:
+        case VKD3DSIH_LOOP:
+        case VKD3DSIH_SWITCH:
```

Comment:
I think this needs all the HS phases as well?

Technically `LABEL` too but we don't support that at the moment.

Reply:
Don't think HS phases are needed. If we are in this situation we are considered masked until the switch/loop situation is popped. HS phases don't generate their own logical scopes per-se I think.